RiotSwiftUI/Modules/Room/PollEditForm/View/PollEditForm.swift

```diff
@@ -0,0 +1,153 @@
+                            .foregroundColor(theme.colors.primaryContent)
+                        
+                        ForEach(0..<viewModel.answerOptions.count, id: \.self) { index in
+                            SafeBindingCollectionEnumerator($viewModel.answerOptions, index: index) { binding in
+                                AnswerOptionGroup(binding: binding.text, index: index) {
+                                    viewModel.send(viewAction: .deleteAnswerOption(viewModel.answerOptions[index]))
+                                }
+                            }
+                        }
+                        
+                        Button(VectorL10n.pollEditFormAddOption) {
+                            viewModel.send(viewAction: .addAnswerOption)
+                        }
+                        .disabled(!viewModel.viewState.addAnswerOptionButtonEnabled)
+                        
+                        Spacer()
+                        
+                        Button(VectorL10n.pollEditFormCreatePoll) {
+                            viewModel.send(viewAction: .create)
+                        }
+                        .buttonStyle(PrimaryActionButtonStyle(enabled: viewModel.viewState.confirmationButtonEnabled))
+                        .disabled(!viewModel.viewState.confirmationButtonEnabled)
+                    }
+                    .animation(.easeInOut(duration: 0.2))
+                    .padding()
+                    .frame(minHeight: proxy.size.height) // Make the VStack fill the ScrollView's parent
+                    .modifier(NavigationTitleBar(titleColor: theme.colors.primaryContent) {
+                        viewModel.send(viewAction: .cancel)
+                    })
+                }
+            }
+        }
+        .accentColor(theme.colors.accent)
+    }
+}
+
+@available(iOS 14.0, *)
+private struct AnswerOptionGroup: View {
+    
+    @Environment(\.theme) private var theme: ThemeSwiftUI
+    
+    @State private var focused = false
+    
+    var binding: Binding<String>
+    
+    let index: Int
+    let onDelete: () -> Void
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 8.0) {
+            Text(VectorL10n.pollEditFormOptionNumber(index + 1))
+                .font(theme.fonts.subheadline)
+                .foregroundColor(theme.colors.primaryContent)
+            
+            HStack(spacing: 16.0) {
+                TextField(VectorL10n.pollEditFormInputPlaceholder, text: binding, onEditingChanged: { edit in
+                    self.focused = edit
+                })
+                .textFieldStyle(BorderedInputFieldStyle(theme: _theme, isEditing: focused))
+                Button {
+                    onDelete()
+                } label: {
+                    Image(uiImage:Asset.Images.pollDeleteOptionIcon.image)
+                }
+            }
+        }
+    }
+}
+
+@available(iOS 14.0, *)
+private struct NavigationTitleBar: ViewModifier {
```

Comment:
Do we need this ViewModifier? Its name is pretty broad considering internals are poll specific.
Can we just put its contents inline in the PollEditForm?

Reply:
We could but I wanted to separate the components a bit. Do you think it's a problem as long as it's private?

Reply:
Fair enough, I'll get rid of it 👍